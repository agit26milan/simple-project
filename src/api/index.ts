import axios from 'axios';

const baseURL = 'https://auntieanne-demo.proseller.io/product/api';

export const getListMenu = async (id: string) => {
  const url = `/productpreset/loadcategory/webOrdering/${id}`;
  return await axios({method: 'get', baseURL, url});
};

export const getSublistMenu = async (companyId: string, id: string) => {
  const url = `/productpreset/loaditems/webOrdering/${companyId}/${id}`;
  return await axios({method: 'post', baseURL, url});
};
