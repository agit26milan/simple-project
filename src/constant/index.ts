export const COMPANY_ID = '64e4e47b-b428-4ee3-97f5-62bfa03c8ba6';
