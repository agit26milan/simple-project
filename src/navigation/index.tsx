import * as React from 'react';
import {createStackNavigator} from '@react-navigation/stack';
import Tab from './TopTab';
import FavoriteScreen from '../screen/Favorite';

const Stack = createStackNavigator();

const RootNavigation = () => {
  return (
    <Stack.Navigator>
      <Stack.Screen
        options={{headerShown: false}}
        name="BottomTab"
        component={Tab}
      />
      <Stack.Screen name="Favorite" component={FavoriteScreen} />
    </Stack.Navigator>
  );
};

export default RootNavigation;
