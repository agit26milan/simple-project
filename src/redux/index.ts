import {configureStore} from '@reduxjs/toolkit';
import {useDispatch, useSelector, TypedUseSelectorHook} from 'react-redux';
import rootReducer from './reducer';
import {persistStore, persistReducer} from 'redux-persist';
import AsyncStorage from '@react-native-async-storage/async-storage';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  whitelist: ['favorite'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
  devTools: __DEV__,
});

export const persistor = persistStore(store);
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
export type AppStore = typeof store;
// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
