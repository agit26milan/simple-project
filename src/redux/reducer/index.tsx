import {combineReducers} from '@reduxjs/toolkit';
import FavoriteReducer from './favorite/reducer';
const appReducer = combineReducers({
  favorite: FavoriteReducer,
});
const rootReducer = (state: any, action: any) => {
  // Clear all data in redux store to initial.

  return appReducer(state, action);
};

export default rootReducer;
