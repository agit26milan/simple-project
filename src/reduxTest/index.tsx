import React from 'react';
import {Provider} from 'react-redux';

type RedusProviderProps = {
  children: React.ReactNode;
  reduxStore: any;
};

const ReduxProvider: React.FC<RedusProviderProps> = ({
  children,
  reduxStore,
}) => <Provider store={reduxStore}>{children}</Provider>;

export default ReduxProvider;
