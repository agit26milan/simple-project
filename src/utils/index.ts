export const formatMoney = (value?: string) => {
  if (value) {
    return value.replace(/(\d)(?=(\d{3})+$)/g, '$1.');
  }
  return 0;
};
