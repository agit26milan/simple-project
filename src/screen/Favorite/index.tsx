import React from 'react';
import {View, StyleSheet, FlatList} from 'react-native';
import {useSelector} from 'react-redux';
import {RootState} from '../../redux';
import Product from '../Home/Product';
import ModalProduct from '../Home/ModalProduct';
import useFavorite from '../../hooks/useFavorite';

const FavoriteScreen = () => {
  const listFavorite = useSelector((state: RootState) => state.favorite.data);
  const [openModal, setOpenModal] = React.useState<boolean>(false);
  const {onSaveFavorite, onRemoveFavorite} = useFavorite();
  const [selectProduct, setSelectProduct] =
    React.useState<Home.ResponseSubMenu | null>(null);

  const onClickProduct = (product: Home.ResponseSubMenu) => {
    setOpenModal(true);
    setSelectProduct(product);
  };

  const onSaveFavoriteHandle = () => {
    if (selectProduct) {
      onSaveFavorite(selectProduct);
    }
  };

  const renderItem = ({item}: any) => (
    <Product item={item} numColumn={3} onPress={onClickProduct} />
  );

  const closeModal = () => {
    setOpenModal(false);
  };

  return (
    <View style={styles.containerMenu}>
      <FlatList numColumns={3} data={listFavorite} renderItem={renderItem} />
      <ModalProduct
        isVisible={openModal}
        onBackButtonPress={closeModal}
        onBackdropPress={closeModal}
        onSaveFavoriteHandle={onSaveFavoriteHandle}
        selectProduct={selectProduct}
        hideFavoriteBtn={true}
        onRemoveFavorite={onRemoveFavorite}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  containerMenu: {
    paddingTop: 20,
    backgroundColor: 'white',
    flex: 1,
  },
});

export default FavoriteScreen;
