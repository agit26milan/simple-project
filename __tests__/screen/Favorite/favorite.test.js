import React from 'react';
import {render} from '@testing-library/react-native';
import Favorite from '../../../src/screen/Favorite';
import ReduxProvider from '../../../src/reduxTest';
import {store} from '../../../src/redux';

jest.mock('redux-persist', () => {
  const real = jest.requireActual('redux-persist');
  return {
    ...real,
    persistReducer: jest
      .fn()
      .mockImplementation((config, reducers) => reducers),
  };
});

describe('Favorite should be match with snapshot', () => {
  const wrapper = ({children}) => (
    <ReduxProvider reduxStore={store}>{children}</ReduxProvider>
  );
  it('Code should be not change', () => {
    const {toJSON} = render(<Favorite />, {wrapper});
    expect(toJSON).toMatchSnapshot();
  });
});
