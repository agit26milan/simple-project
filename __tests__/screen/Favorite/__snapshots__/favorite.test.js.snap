// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite should be match with snapshot Code should be not change 1`] = `[Function]`;

exports[`it should be match with snapshot Code should be not change 1`] = `[Function]`;
